pombola/kenya/static/js/facebook-experiment.js

```diff
@@ -0,0 +1,170 @@
+    $('#ctas').show();
+
+    var options = {
+            placeholder: 'Constituency',
+            allowClear: true,
+        },
+
+        modelMatcher = function (term, text) {
+            /* From https://github.com/select2/select2/issues/3034#issuecomment-126451444 */
+            text.parentText = text.parentText || "";
+
+            // Always return the object if there is nothing to compare
+            if ($.trim(term.term) === '') {
+                return text;
+            }
+
+        // Do a recursive check for options with children
+        if (text.children && text.children.length > 0) {
+            // Clone the text object if there are children
+            // This is required as we modify the object to remove any non-matches
+            var match = $.extend(true, {}, text);
+
+            // Check each child of the option
+            for (var c = text.children.length - 1; c >= 0; c--) {
+                var child = text.children[c];
+                child.parentText += text.parentText + " " + text.text;
+
+                var matches = modelMatcher(term, child);
+
+                // If there wasn't a match, remove the object in the array
+                if (matches == null) {
+                    match.children.splice(c, 1);
+                }
+            }
+
+            // If any children matched, return the new object
+            if (match.children.length > 0) {
+                return match;
+            }
+
+            // If there were no matching children, check just the plain object
+            return modelMatcher(term, match);
+        }
+
+      // If the typed-in term matches the text of this term, or the text from any
+      // parent term, then it's a match.
+      var original = (text.parentText + ' ' + text.text).toUpperCase();
+      var term = term.term.toUpperCase();
+
+
+      // Check if the text contains the term
+      if (original.indexOf(term) > -1) {
+        return text;
+      }
+
+      // If it doesn't contain the term, don't return anything
+      return null;
+    }
+
+    var matcherObject = {matcher: modelMatcher};
+    var optionsWithGroupMatcherNarrow = $.extend({}, matcherObject, options);
+    var iebcSelector = $('.iebc-office-lookup select');
+
+    // Do the IEBC office selector magic
+    iebcSelector.select2(optionsWithGroupMatcherNarrow);
+    iebcSelector.on("select2:select", function(e) {
+        e.preventDefault();
+
+        $.ajax({
+            url: "/iebc-office-ajax",
+            data: {
+                area: iebcSelector.val()
+            },
+            type: "GET",
+            dataType: "html",
+            success: function (data) {
+                $('#constituency-lookup-result').html(data);
+            },
+            error: function (xhr, status) {
+                $('#constituency-lookup-result').html('<p>Sorry, something went wrong!</p>');
+            }
+        });
+
+        // Do we have a SID?
+        if (sid = getParameterByName('sid')) {
```

Comment:
Is `sid` meant to be a global - I'm assuming not, so there should be a `var sid` at the start of this function.